scripts/lib/upgrade-zulip-from-git

```diff
@@ -90,11 +146,23 @@ try:
     )
 
     # Generate the deployment directory via git worktree from our local repository.
-    commit_hash = subprocess.check_output(
-        ["git", "rev-parse", refname],
-        preexec_fn=su_to_zulip,
-        text=True,
-    ).strip()
+    try:
+        commit_hash = subprocess.check_output(
+            ["git", "rev-parse", "--verify", refname],
```

Comment:
I think we want to restrict this first call to looking at `f"refs/tags/{refname}"`, to avoid preferring a stale local branch `main` over a freshly fetched remote branch `origin/main` in case someone previously ran `git checkout main`.

(If someone really wants to trigger an upgrade to a local branch, they can use `--remote-url=.` or `--remote-url=/srv/zulip.git` or something.)

Reply:
Good point -- updated.